rss2email/feed.py

```diff
@@ -513,31 +534,43 @@ def _process_entry(self, parsed, entry):
             extra_headers=extra_headers,
             config=self.config,
             section=self.section)
-        return (guid, guid, sender, message)
 
-    def _get_entry_id(self, entry):
-        """Get best ID from an entry."""
+        return guid, new_state, sender, message
+
+    def _get_uid_for_entry(self, entry) -> str:
+        """Get the best UID (unique ID) for the entry."""
         if self.trust_link:
-            return entry.get('link', None)
+            uid = self._get_entry_link(entry)
+            if uid:
+                return uid
         if self.trust_guid:
-            if getattr(entry, 'id', None):
-                # Newer versions of feedparser could return a dictionary
-                if isinstance(entry.id, dict):
-                    return entry.id.values()[0]
-                return entry.id
+            uid = self._get_entry_id(entry)
+            if uid:
+                return uid
+        return self._get_entry_hash(entry)
+
+    @lru_cache(1)
```

Comment:
This is fine, but I'm wondering why you put this here. Is it the case that we're calling `_get_entry_hash` on the same entry repeatedly, so that an LRU cache of size 1 is useful? I haven't examined the usages of this function in detail so that might very well be the case, I'm just curious.

Reply:
Yes, it is just such a case. The `_get_entry_hash` method may be called twice:
1. [`_process_entry`](https://github.com/rss2email/rss2email/blob/9d7cb815a212c607f3b2a5e33bbabfe3f892000a/rss2email/feed.py#L472) > [`_get_uid_for_entry`](https://github.com/rss2email/rss2email/blob/9d7cb815a212c607f3b2a5e33bbabfe3f892000a/rss2email/feed.py#L550) > `_get_entry_hash`
2. [`_process_entry`](https://github.com/rss2email/rss2email/blob/9d7cb815a212c607f3b2a5e33bbabfe3f892000a/rss2email/feed.py#L473) > `_get_entry_hash`

Also the `_get_entry_content` method may be called several times:
1. [`_get_entry_hash`](https://github.com/rss2email/rss2email/blob/9d7cb815a212c607f3b2a5e33bbabfe3f892000a/rss2email/feed.py#L554) > `_get_entry_content`
2. [`_get_entry_title`](https://github.com/rss2email/rss2email/blob/9d7cb815a212c607f3b2a5e33bbabfe3f892000a/rss2email/feed.py#L582) > `_get_entry_content`
3. [`_process_entry`](https://github.com/rss2email/rss2email/blob/9d7cb815a212c607f3b2a5e33bbabfe3f892000a/rss2email/feed.py#L521) > `_get_entry_content`